File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/detail/NetsOneContract.java
package xuk.imaginary.gui.nets.detail;

import xuk.imaginary.BasePresenter;
import xuk.imaginary.BaseView;
import xuk.imaginary.data.NetsDetail;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface NetsOneContract {

    interface View extends BaseView<Presenter> {
        void refreshNewsOne(NetsDetail netsDetail);
    }

    interface Presenter extends BasePresenter {
        void getNewsOneRequest(String postId);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/detail/NetsOneContract.kt
package xuk.imaginary.gui.nets.detail

import xuk.imaginary.BasePresenter
import xuk.imaginary.BaseView
import xuk.imaginary.data.NetsDetail

/**
 * This specifies the contract between the view and the presenter.
 */
interface NetsOneContract {

  interface View : BaseView<Presenter> {
    fun refreshNewsOne(netsDetail: NetsDetail)
  }

  interface Presenter : BasePresenter {
    fun getNewsOneRequest(postId: String)
  }
}
